DataFormats/TrackingRecHitSoA/test/alpaka/Hits_test.cc

```diff
@@ -34,33 +37,68 @@ int main() {
+
+      auto hitsX = cms::alpakatools::make_host_buffer<float[]>(queue, nHits);
+      for (size_t i = 0; i < nHits; ++i) {
+        hitsX[i] = float(i) * 2;
+      }
+
+      auto moduleStartD = cms::alpakatools::make_device_view<uint32_t>(queue, clusters.view().clusModuleStart(), nHits);
       alpaka::memcpy(queue, moduleStartD, moduleStartH);
-      TrackingRecHitsSoACollection<pixelTopology::Phase1> tkhit(queue, nHits, offset, moduleStartD.data());
 
-      testTrackingRecHitSoA::runKernels<pixelTopology::Phase1>(tkhit.view(), queue);
+      TrackingRecHitsSoACollection tkhit(queue, clusters);
+
+      // exercise the copy of a full column (on device)
+      auto hitXD = cms::alpakatools::make_device_view<float>(queue, tkhit.view().xLocal(), nHits);
+      alpaka::memcpy(queue, hitXD, hitsX);
+
+      // exercise the memset of a colum (on device)
+      auto hitYD = cms::alpakatools::make_device_view<float>(queue, tkhit.view().yGlobal(), nHits);
+      constexpr float constYG = -14.0458;
+      std::vector<float> constYV(nHits, constYG);
+      auto constYGV_v = cms::alpakatools::make_host_view<float>(constYV.data(), nHits);
+      alpaka::memcpy(queue, hitYD, constYGV_v);
+
+      testTrackingRecHitSoA::runKernels(tkhit.view(), tkhit.view<::reco::HitModuleSoA>(), queue);
       tkhit.updateFromDevice(queue);
 
 #if defined ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED or defined ALPAKA_ACC_CPU_B_TBB_T_SEQ_ENABLED
       // requires c++23 to make cms::alpakatools::CopyToHost compile using if constexpr
       // see https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p2593r0.html
-      TrackingRecHitHost<pixelTopology::Phase1> const& host_collection = tkhit;
-      // wait for the kernel to complete
-      alpaka::wait(queue);
+      ::reco::TrackingRecHitHost const& host_collection = tkhit;
 #else
-      using CopyT = cms::alpakatools::CopyToHost<TrackingRecHitDevice<pixelTopology::Phase1, Device> >;
-      TrackingRecHitHost<pixelTopology::Phase1> host_collection = CopyT::copyAsync(queue, tkhit);
-      // wait for the kernel and the copy to complete
-      alpaka::wait(queue);
-      CopyT::postCopy(host_collection);
+      ::reco::TrackingRecHitHost host_collection =
+          cms::alpakatools::CopyToHost<::reco::TrackingRecHitDevice<Device>>::copyAsync(queue, tkhit);
 #endif
 
+      alpaka::wait(queue);
+
+      alpaka::QueueCpuBlocking queue_host{cms::alpakatools::host()};
+
+      ::reco::TrackingRecHitHost host_collection_2(cms::alpakatools::host(), nHits, nModules);
+
+      // exercise the memset of a colum (on host)
+      auto hitLYH = cms::alpakatools::make_host_view<float>(host_collection_2.view().yLocal(), nHits);
+      constexpr float constYL = -27.0855;
+      std::vector<float> constYLV(nHits, constYL);
+      auto constYL_v = cms::alpakatools::make_host_view<float>(constYLV.data(), nHits);
+      alpaka::memcpy(queue_host, hitLYH, constYL_v);
+      // wait for the kernel and the potential copy to complete
```

Comment:
either add
```c++
alpaka::wait(queue_host);
```
or remove the comment

Reply:
Would it be useful in this case to have this wait with the host queue?

Reply:
Probably not: today the host queue is blocking, so the wait is not needed (but also is very cheap).
If we later add a non-blocking host queue, it will likely need various updates anyway.